teuthology/task/install/rpm.py

```diff
@@ -18,49 +17,54 @@ def _remove(ctx, config, remote, rpm):
     :param remote: the teuthology.orchestra.remote.Remote object
     :param rpm: list of packages names to remove
     """
-    rpm = _package_overrides(rpm, remote.os)
+    remote_os = remote.os
+    dist_release = remote_os.name
+    rpm = _package_overrides(rpm, remote_os)
     log.info("Removing packages: {pkglist} on rpm system.".format(
         pkglist=", ".join(rpm)))
-    builder = _get_builder_project(ctx, remote, config)
-    dist_release = builder.dist_release
+    repos = config.get('repos')
 
     if dist_release in ['opensuse', 'sle']:
         pkg_mng_cmd = 'zypper'
-        pkg_mng_opts = '-n --no-gpg-checks'
-        pkg_mng_subcommand_opts = '--capability'
+        pkg_mng_opts = ['-n', '--no-gpg-checks']
+        pkg_mng_subcommand_opts = ['--capability']
     else:
         pkg_mng_cmd = 'yum'
-        pkg_mng_opts = '-y'
-        pkg_mng_subcommand_opts = ''
+        pkg_mng_opts = ['-y']
+        pkg_mng_subcommand_opts = []
 
     remote.run(
         args=[
-            'for', 'd', 'in',
-        ] + rpm + [
-            run.Raw(';'),
-            'do',
-            'sudo',
-            pkg_mng_cmd, pkg_mng_opts, 'remove', pkg_mng_subcommand_opts,
-            run.Raw('$d'), run.Raw('||'), 'true', run.Raw(';'),
-            'done',
+            'for', 'd', 'in' ] + rpm + [ run.Raw(';'), 'do',
+                'sudo', pkg_mng_cmd ] + pkg_mng_opts + ['remove'] +
+                pkg_mng_subcommand_opts + [ run.Raw('$d'),
+                run.Raw('||'), 'true',
+                run.Raw(';'), 'done',
         ])
     if dist_release in ['opensuse', 'sle']:
-        pkg_mng_opts = '-a'
+        pkg_mng_opts = ['-a']
     else:
-        pkg_mng_opts = 'all'
+        pkg_mng_opts = ['all']
     remote.run(
-        args=[
-            'sudo', pkg_mng_cmd, 'clean', pkg_mng_opts,
-        ])
+        args=['sudo', pkg_mng_cmd, 'clean'] + pkg_mng_opts
+    )
 
-    builder.remove_repo()
+    if repos:
+        if dist_release in ['opensuse', 'sle']:
+            _zypper_removerepo(remote, repos)
+        else:
+            # _yum_removerepo(repos)
```

Comment:
please either drop the commented-out line, or add a `# _yum_addrepo(repos)` line to the similar conditional in the addrepo method